src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -252,6 +252,72 @@ void bulk_load_service::create_partition_bulk_load_dir(const std::string &app_na
 
 // ThreadPool: THREAD_POOL_META_STATE
 void bulk_load_service::partition_bulk_load(const std::string &app_name, const gpid &pid)
+{
+    FAIL_POINT_INJECT_F("meta_bulk_load_partition_bulk_load", [](dsn::string_view) {});
+
+    rpc_address primary_addr;
+    ballot b;
+    {
+        zauto_read_lock l(app_lock());
+        std::shared_ptr<app_state> app = _state->get_app(pid.get_app_id());
+        if (app == nullptr || app->status != app_status::AS_AVAILABLE) {
+            dwarn_f("app(name={}, id={}) is not existed, set bulk load failed",
+                    app_name,
+                    pid.get_app_id());
+            handle_app_unavailable(pid.get_app_id(), app_name);
+            return;
+        }
+        primary_addr = app->partitions[pid.get_partition_index()].primary;
+        b = app->partitions[pid.get_partition_index()].ballot;
+    }
+
+    if (primary_addr.is_invalid()) {
+        dwarn_f("app({}) partition({}) primary is invalid, try it later", app_name, pid);
+        tasking::enqueue(LPC_META_STATE_NORMAL,
+                         _meta_svc->tracker(),
+                         std::bind(&bulk_load_service::partition_bulk_load, this, app_name, pid),
+                         0,
+                         std::chrono::seconds(1));
+        return;
+    }
+
+    zauto_read_lock l(_lock);
+    const app_bulk_load_info &ainfo = _app_bulk_load_info[pid.get_app_id()];
```

Comment:
ainfo -> bulk_load_info?

Reply:
`_app_bulk_load_info` is a map: <app_id, app_bulk_load_info>.
The type of ainfo is app_bulk_load_info

Reply:
Well, I just think the `ainfo` is simple